### PR TITLE
nova: remove/rename options deprecated in ocata

### PR DESCRIPTION
**Pull request by stefannica**
The following deprecated nova configuration options
are moved, removed or renamed in this PR:
    

- [default]/verbose deprecated option
- [keystone_authtoken]/service_token_roles_required is set to True to enable fetching expired tokens based on valid service tokens
- auth_strategy and vendordata_jsonfile_path moved from DEFAULT group to api group
- scheduler_host_manager moved from DEFAULT group to scheduler group and renamed to host_manager
- scheduler_use_baremetal_filters, scheduler_available_filters, scheduler_default_filters and scheduler_tracks_instance_changes moved from DEFAULT group to filter_scheduler group and renamed to use_baremetal_filters, available_filters, enabled_filters and track_instance_changes respectively
- notify_on_state_change moved from DEFAULT group to the notifications group
- removed deprecated dhcp_domain and use_neutron config options
    
This PR also removes the nova certificates API bits; nova-cert has been
deprecated in Newton and will be removed in Pike.

Not included in this PR:
- the deprecated keystonemiddleware.auth_token in-process token needs to be replaced with a memcache server (done separately, due to security implications - see #1090)

References:
- Ocata release notes for nova: https://docs.openstack.org/releasenotes/nova/ocata.html
- Ocata release notes for keystonemiddleware: https://docs.openstack.org/releasenotes/keystonemiddleware/ocata.html
- nova specification removing nova-cert: https://specs.openstack.org/openstack/nova-specs/specs/pike/approved/remove-nova-cert.html
- Newton release notes for nova: https://docs.openstack.org/releasenotes/nova/newton.html

### CODE REVIEW

**Comment by stefannica** [comment]
One of the changes in this PR seems to be breaking the nova placement API _in develcloud8 jobs_, but not in ocatacloud8 jobs, where Ocata packages are used. Need to investigate this further.

**Comment by stefannica** [comment]
The nova placement API errors in develcloud8 are caused by the missing file /srv/www/nova-placement-api/api.wsgi, which is installed by the ocata openstack-nova package, but not by the neutron package.
The nova-placement service has been configured as a wsgi frontend in #1057, but the neutron package hasn't yet been updated to install the /usr/bin/nova-placement-api as the api.wsgi file.

What I don't understand is how this has worked in develcloud8 before this PR. @toabctl, maybe you could help me understand what I've done to trigger this ?

**Comment by toabctl** [comment]
@stefannica 
- why should the neutron package install something related to nova-placement-api?
- the error I see in the nova-compute.log is:
```
2017-07-27 13:32:32.300 1677 ERROR nova.scheduler.client.report [req-85e6bd27-4dd8-4388-90fc-263311a2302a - - - - -] Failed to retrieve resource provider record from placement 
API for UUID f8161a61-cda5-44e1-ad60-f24399b3fa94. Got 403: <?xml version="1.0" encoding="UTF-8"?>
...
Access forbidden!
...
```

**Comment by dirkmueller** [comment]
@stefannica requires rebase

**Comment by stefannica** [comment]
@toabctl my apologies, I meant the nova package, of course (I updated the comment). I'm talking about the difference between https://build.opensuse.org/package/view_file/Cloud:OpenStack:Newton/openstack-nova/openstack-nova.spec?expand=1
and https://build.opensuse.org/package/view_file/Cloud:OpenStack:Ocata/openstack-nova/openstack-nova.spec?expand=1. The '/srv/www/%{component}-placement-api/app.wsgi' parts are missing in the newton package.

**Comment by stefannica** [comment]
The mistery behind the failed builds is actually quite simple: nova-network was enabled by default when the deprecated use_neutron option was removed. This option still needs to be kept around until the switch to ocata packages. @toabctl sorry for dragging you into this, the nova-placement api errors were just along for the ride, completely not related.

**Comment by stefannica** [comment]
@toabctl actually that commit is no longer needed since the ocatacloud8->develcloud8 merge. Thanks for the reminder, I removed it.

File: chef/cookbooks/nova/templates/default/nova.conf.erb [hunk 1]
```diff
@@ -42,6 +42,7 @@ use_cow_images = false
 <% end -%>
 firewall_driver = nova.virt.firewall.NoopFirewallDriver
 <%= "ssl_ca_file = #{@ssl_ca_file}" if @ssl_cert_required %>
+
```
Comment (toabctl) [inline chef/cookbooks/nova/templates/default/nova.conf.erb]: unrelated change
